src/ansys/speos/core/sensor.py

```diff
@@ -3742,3 +3742,617 @@ def set_geometries(self, geometries: [List[GeoRef]]) -> Sensor3DIrradiance:
+        """
+        if isinstance(self._type, BaseSensor.Spectral):
+            return self._type
+        else:
+            return None
+
+    @property
+    def layer(
+        self,
+    ) -> Union[
+        None,
+        SensorIrradiance,
+        BaseSensor.LayerTypeFace,
+        BaseSensor.LayerTypeSequence,
+        BaseSensor.LayerTypeIncidenceAngle,
+    ]:
+        """Property containing all options in regard to the layer separation properties.
+
+        Returns
+        -------
+        Union[\
+            None,\
+            ansys.speos.core.sensor.SensorIrradiance,\
+            ansys.speos.core.sensor.BaseSensor.LayerTypeFace,\
+            ansys.speos.core.sensor.BaseSensor.LayerTypeSequence,\
+            ansys.speos.core.sensor.BaseSensor.LayerTypeIncidenceAngle\
+        ]
+            Instance of Layertype Class for this sensor feature
+        """
+        return self._layer_type
+
+    def set_orientation_x_as_meridian(self):
+        """Set Orientation type: X As Meridian, Y as Parallel."""
+        self._sensor_template.intensity_sensor_template.intensity_orientation_x_as_meridian.SetInParent()
+        self._set_default_dimension_values()
+
+    def set_orientation_x_as_parallel(self):
+        """Set Orientation type: X as Parallel, Y As Meridian."""
+        self._sensor_template.intensity_sensor_template.intensity_orientation_x_as_parallel.SetInParent()
+        self._set_default_dimension_values()
+
+    def set_orientation_conoscopic(self):
+        """Set Orientation type: Conoscopic."""
+        self._sensor_template.intensity_sensor_template.intensity_orientation_conoscopic.SetInParent()
+        self._set_default_dimension_values()
+
+    def set_viewing_direction_from_source(self):
+        """Set viewing direction from Source Looking At Sensor."""
+        self._sensor_template.intensity_sensor_template.from_source_looking_at_sensor.SetInParent()
+
+    def set_viewing_direction_from_sensor(self):
+        """Set viewing direction from Sensor Looking At Source."""
+        self._sensor_template.intensity_sensor_template.from_sensor_looking_at_source.SetInParent()
+
+    def _set_default_dimension_values(self):
+        template = self._sensor_template.intensity_sensor_template
+        if template.HasField("intensity_orientation_conoscopic"):
+            self.theta_max = 45
+            self.theta_max_sampling = 90
+        elif template.HasField("intensity_orientation_x_as_parallel"):
+            self.x_start = -30
+            self.x_end = 30
+            self.x_sampling = 120
+            self.y_start = -45
+            self.y_end = 45
+            self.y_sampling = 180
+        elif template.HasField("intensity_orientation_x_as_meridian"):
+            self.y_start = -30
+            self.y_end = 30
+            self.y_sampling = 120
+            self.x_start = -45
+            self.x_end = 45
+            self.x_sampling = 180
+
+    @property
+    def x_start(self) -> float:
```

Comment:
then x_start will be shown in the auto-completion methods list even the orientation is set as conoscopic....

Reply:
yes butt you can't set it and it will return non on conoscopic mode